lib/std/c/darwin.zig

```diff
@@ -1513,3 +1515,178 @@ pub const DB_RECORDTYPE = enum(u32) {
     pub const APP_DEFINED_START = 0x80000000;
     pub const APP_DEFINED_END = 0xffffffff;
 };
+
+pub fn getKernError(err: std.c.kern_return_t) KernE {
+    return @as(KernE, @enumFromInt(@as(u32, @truncate(@as(usize, @intCast(err))))));
+}
+
+pub fn unexpectedKernError(err: KernE) std.posix.UnexpectedError {
+    if (std.posix.unexpected_error_tracing) {
+        std.debug.print("unexpected error: {d}\n", .{@intFromEnum(err)});
+        std.debug.dumpCurrentStackTrace(null);
+    }
+    return error.Unexpected;
+}
+
+/// Kernel return values
+pub const KernE = enum(u32) {
```

Comment:
according to kern_return.h this should be 
```suggestion
pub const KERN = enum(u32) {
```

Reply:
will do. how do you deal with prefix collision between different types? such as `MachMsgE` (a bit further) for instance? https://github.com/ziglang/zig/pull/21218/files#diff-f69178bf0db31205e046265d33d0eff15e5ddfdce2a7a18dc39309db4da64a5eR9674

Reply:
`MachMsgE` is a made up name that doesn't seem to be based on any C bits, so that's a problem. If it's so messy that nonexhaustive enums and packed structs cannot be used, then the best move would be to give up and use a single `MACH` empty struct namespace for all of them. It is not obvious to me based on a quick glance how these bits relate to each other.